guides/source/active_support_instrumentation.md

```diff
@@ -3,623 +3,1160 @@
 Active Support Instrumentation
 ==============================
 
-Active Support is a part of core Rails that provides Ruby language extensions, utilities, and other things. One of the things it includes is an instrumentation API that can be used inside an application to measure certain actions that occur within Ruby code, such as those inside a Rails application or the framework itself. It is not limited to Rails, however. It can be used independently in other Ruby scripts if desired.
+[Active Support](https://api.rubyonrails.org/classes/ActiveSupport.html) is a
+core component of Ruby on Rails, offering a wide range of Ruby language
+extensions and utility classes.
 
-In this guide, you will learn how to use the Active Support's instrumentation API to measure events inside of Rails and other Ruby code.
+One of its key features is the Instrumentation API, which is described in
+further detail in this guide. You will learn:
 
-After reading this guide, you will know:
-
-* What instrumentation can provide.
-* How to add a subscriber to a hook.
-* The hooks inside the Rails framework for instrumentation.
-* How to build a custom instrumentation implementation.
+* About Instrumentation
+* How to instrument your own events.
+* How to subscribe to events.
+* The instrumentation hooks available inside Rails.
 
 --------------------------------------------------------------------------------
 
 Introduction to Instrumentation
 -------------------------------
 
-The instrumentation API provided by Active Support allows developers to provide hooks which other developers may hook into. There are [several of these](#rails-framework-hooks) within the Rails framework. With this API, developers can choose to be notified when certain events occur inside their application or another piece of Ruby code.
+The Instrumentation API provides a way to instrument code and subscribe to
+events that occur in your application. Instrumentation means wrapping a block of
+code so that, when it runs, an event is emitted with a name and optional
+payload. Any subscribers listening for that event will then be notified and can
+react, for example, they can log information, benchmark, or perform some other
+action. This makes it possible to observe behavior within the Rails framework,
+in your own code, or even in standalone Ruby scripts. There are a few parts that
+are vital to understanding the Instrumentation API: hooks, events, and
+subscribers.
+
```

Comment:
We should probably have a NOTE here to make clear that instrumentation should be used for framework code and not application code, and link to "Active Support Event Reporter" for that instead.